library/src/main/java/org/apache/apex/malhar/lib/window/impl/SpillableWindowedPlainStorage.java

```diff
@@ -0,0 +1,188 @@
+import org.apache.apex.malhar.lib.window.WindowedStorage;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.netlet.util.Slice;
+
+/**
+ * This is an implementation of WindowedPlainStorage that makes use of {@link Spillable} data structures
+ *
+ * @param <T> Type of the value per window
+ */
+public class SpillableWindowedPlainStorage<T> implements WindowedStorage.WindowedPlainStorage<T>
+{
+  private SpillableStateStore store;
+  private transient SpillableComplexComponentImpl sccImpl;
+  private long bucket;
+  private Serde<Window, Slice> windowSerde;
+  private Serde<T, Slice> valueSerde;
+
+  protected transient Spillable.SpillableByteMap<Window, T> internMap;
+
+  public SpillableWindowedPlainStorage()
+  {
+  }
+
+  public SpillableWindowedPlainStorage(long bucket, Serde<Window, Slice> windowSerde, Serde<T, Slice> valueSerde)
+  {
+    this.bucket = bucket;
+    this.windowSerde = windowSerde;
+    this.valueSerde = valueSerde;
+  }
+
+  public void setStore(SpillableStateStore store)
+  {
+    this.store = store;
+  }
+
+  public void setBucket(long bucket)
+  {
+    this.bucket = bucket;
+  }
+
+  public void setWindowSerde(Serde<Window, Slice> windowSerde)
+  {
+    this.windowSerde = windowSerde;
+  }
+
+  public void setValueSerde(Serde<T, Slice> valueSerde)
+  {
+    this.valueSerde = valueSerde;
+  }
+
+  @Override
+  public void put(Window window, T value)
+  {
+    internMap.put(window, value);
+  }
+
+  @Override
+  public T get(Window window)
+  {
+    return internMap.get(window);
+  }
+
+  @Override
+  public Iterable<Map.Entry<Window, T>> entrySet()
+  {
+    return internMap.entrySet();
+  }
+
+  @Override
+  public Iterator<Map.Entry<Window, T>> iterator()
+  {
+    return internMap.entrySet().iterator();
+  }
+
+  @Override
+  public boolean containsWindow(Window window)
+  {
+    return internMap.containsKey(window);
+  }
+
+  @Override
+  public long size()
+  {
+    return internMap.size();
+  }
+
+  @Override
+  public void remove(Window window)
+  {
+    internMap.remove(window);
+  }
+
+  @Override
+  public void migrateWindow(Window fromWindow, Window toWindow)
+  {
+    internMap.put(toWindow, internMap.remove(fromWindow));
+  }
+
+  @Override
+  public void setup(Context.OperatorContext context)
+  {
+    if (store == null) {
+      // provide a default store
+      store = new ManagedStateSpillableStateStore();
+    }
+    if (bucket == 0) {
+      // choose a bucket that is almost guaranteed to be unique
+      bucket = (context.getValue(Context.DAGContext.APPLICATION_NAME) + "#" + context.getId()).hashCode();
+    }
+    // set default serdes
+    if (windowSerde == null) {
+      windowSerde = new SerdeKryoSlice<>();
+    }
+    if (valueSerde == null) {
+      valueSerde = new SerdeKryoSlice<>();
+    }
+    sccImpl = new SpillableComplexComponentImpl(store);
+    sccImpl.setup(context);
+    internMap = sccImpl.newSpillableByteMap(bucket, windowSerde, valueSerde);
```

Comment:
@davidyan74 I think you are getting a size of zero because you are allocating a new spillable data structure here. You should check if sccImpl is null and only if it's null initialize it and all the SpillableDatastructures